src/sql/SqlService.ts

```diff
@@ -284,7 +282,7 @@ export class SqlServiceImpl implements SqlService {
         );
     }
 
-    executeStatement(sqlStatement: SqlStatement): SqlResult {
+    executeStatement(sqlStatement: SqlStatement): Promise<SqlResult> {
```

Comment:
After this change, I assume the methods on the SqlResult (updateCount etc.) not have to return Promise, right? 
Should we change those as well? Maybe we can leave them like this for being future-proof, but I am not sure

Reply:
What do you mean by being future-proof? I think we can do that. This will simplify the SqlResult.

Reply:
Since we will always return a promise from this execute()(the one that optionally waits execution response), the user will always wait in order to get a SqlResult object. Then we don't need to return promises from other methods.

Also I don't think we need another execute method in the future(sync or async version) because the user can use this execute method for both async and sync. By async, I mean without waiting the execute response. Therefore we can simplify SqlResult and delete executeDeferred variable altogether. WDYT?

Reply:
I meant if we want to use the same API when we introduce NonBlockingSqlResult in the future, but we could introduce a new type to represent that in the future. So, resolving this